api4/post.go

```diff
@@ -185,6 +185,29 @@ func getPostsForChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 		mlog.Error("Failed to prepare posts for getPostsForChannel response", mlog.Any("err", err))
 	}
 
+	postListLength := len(clientPostList.Order)
+
+	var nextPostId string
+	if c.Params.Page == 0 {
+		nextPostId = beforePost
```

Comment:
There's at least a few cases where this is incorrect. Say you have a channel with posts with IDs `post0` (the oldest) to `post9` (the newest), it won't work in at least the following cases:
1. When page is 0, it'll only set the next and previous post IDs from the before/after parameters, when it needs to get the other one from the database. For example, if you request `?after=post3&page=0&per_page=3`, you'll get
    ```
    {
        next_post_id: "", // This should be post7, but it's empty because we're looking for the `before` parameter
        order: ["post6", "post5", "post4"],
        prev_post_id: "post3"
    }
    ```
    The same thing happens with the previous post ID and `?before=...`.